### PR TITLE
Useropt - Merging all user tables for db optimization

### PR DESCRIPTION
**Pull request by radicalray**
Merged all user, profile, trainee, TA tables into one and used Proxy models to fake the old tables.

 - Created new admin pages to adapt to new schema
 - Ironed out as many wrinkles as I can think of throughout the whole project.
 - Checked functionality (all working) in web access, badges, life study, and exams.
 - Merged active into is_active within users
 - renamed term to terms_attended
 - @haileyl Added helper functions to get trainee from request.user so please use (aputil.utils):
    - trainee_from_user
    - is_trainee (also template filter as well)
    - is_TA

 - [ ] autofixture not working yet
 - [ ] migrations not squashed in a nice way yet. Need to figure out how to cherrypick commit

@schai6 @davidhtien @lifedispenser @ricefield 

Please code review me, Planning to merge tmr.

### CODE REVIEW

File: ap/accounts/models.py [hunk 1]
```diff
@@ -136,99 +169,69 @@ def email_user(self, subject, message, from_email=None):
...
+    terms_attended = models.ManyToManyField(Term)   
+    current_term = models.IntegerField(default=1, null=True, blank=True)
+    
+    date_begin = models.DateField(null=True, blank=True)
     date_end = models.DateField(null=True, blank=True)
 
-    badge = models.ForeignKey(Badge, blank=True, null=True)
-
-    TA = models.ForeignKey(TrainingAssistant, null=True, blank=True)
+    TA = models.ForeignKey('self', related_name='training_assistant', null=True, blank=True)
```
Comment (leonardlan) [inline ap/accounts/models.py]: I'm curious what 'self' means for ForeignKey.

File: ap/accounts/models.py [hunk 2]
```diff
@@ -136,99 +169,69 @@ def email_user(self, subject, message, from_email=None):
...
             if not discipline.is_completed():
                 o_discipline.append(discipline)
         return o_discipline
 
-    email = property(_trainee_email)  # should just use trainee.account.email
+    # Optional meta field to lighten each user object
+    # meta = models.OneToOneField(UserMeta, related_name='user', null=True, blank=True)
+
+
+class TraineeManager(models.Manager):
```
Comment (leonardlan) [inline ap/accounts/models.py]: This is genius. Kudos. :+1:

**Comment by leonardlan** [comment]
:+1:

**Comment by radicalray** [comment]
Merging!

**Pull request merged**